### PR TITLE
Add ability to specify BlockBasedTableOptions for rocksdb

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add ability to specify BlockBasedTableOptions for EmbeddedRocksDB

List of possible options - https://github.com/facebook/rocksdb/blob/4b013dcbed2df84fde3901d7655b9b91c557454d/table/block_based/block_based_table_factory.cc#L228

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 6bc4d83eccda107853bc42c4411ea93359424af3 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56264/6bc4d83eccda107853bc42c4411ea93359424af3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6734044415">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56264/6bc4d83eccda107853bc42c4411ea93359424af3/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/integration/test_rocksdb_options/test.py [hunk 1]
```diff
@@ -149,6 +149,31 @@ def test_invalid_column_family_options(start_cluster):
     node.restart_clickhouse()
 
 
+def test_invalid_block_based_table_options(start_cluster):
+    node.exec_in_container(
+        [
+            "bash",
+            "-c",
+            "sed -i 's/block_size/no_such_block_based_table_options/g' /etc/clickhouse-server/config.d/rocksdb.xml",
```
Comment (vdimir) [inline tests/integration/test_rocksdb_options/test.py:157]: Why not just have a several configs in this tests and start several nodes with different setups?

File: programs/server/config.xml [hunk 2]
```diff
@@ -1472,6 +1472,15 @@
     <!-- <disable_internal_dns_cache>1</disable_internal_dns_cache> -->
 
     <!-- You can also configure rocksdb like this: -->
+    <!-- Full list of options:
+         - options:
+           - https://github.com/facebook/rocksdb/blob/4b013dcbed2df84fde3901d7655b9b91c557454d/include/rocksdb/options.h#L1452
+         - column_family_options:
+           - https://github.com/facebook/rocksdb/blob/4b013dcbed2df84fde3901d7655b9b91c557454d/include/rocksdb/options.h#L66
+         - block_based_table_options:
+           - https://github.com/facebook/rocksdb/blob/4b013dcbed2df84fde3901d7655b9b91c557454d/table/block_based/block_based_table_factory.cc#L228
```
Comment (vdimir) [inline programs/server/config.xml:1481]: Are we advanced users?
```cpp
// For advanced user only
struct BlockBasedTableOptions {
```

File: programs/server/config.xml [hunk 3]
```diff
@@ -1472,6 +1472,15 @@
     <!-- <disable_internal_dns_cache>1</disable_internal_dns_cache> -->
 
     <!-- You can also configure rocksdb like this: -->
+    <!-- Full list of options:
```
Comment (vdimir) [inline programs/server/config.xml:1475]: Why user may want to change table layout? Does it only affect performance?

File: programs/server/config.xml [hunk 4]
```diff
@@ -1480,6 +1489,9 @@
         <column_family_options>
             <num_levels>2</num_levels>
         </column_family_options>
+        <block_based_table_options>
+            <block_size>1024</block_size>
```
Comment (vdimir) [inline programs/server/config.xml:1493]: What will happen if user changes this setting for existing database? Will we get an error in `StorageEmbeddedRocksDB::initDB` or changes are supported by rocks db?

File: programs/server/config.xml (hunk 2 above)
Comment (azat) [reply programs/server/config.xml:1481]: It depends. For instance I want to play with `block_size` and `max_auto_readahead_size`.

File: tests/integration/test_rocksdb_options/test.py (hunk 1 above)
Comment (azat) [reply tests/integration/test_rocksdb_options/test.py:157]: I don't see how running multiple instances better then sed + restart, more instances - likely will eat more resources, and there are problems with stability of integration tests.

File: programs/server/config.xml (hunk 4 above)
Comment (azat) [reply programs/server/config.xml:1493]: No, they will be applied for new parts only, everything will work (and you can do manual compact to force update - #56225 )

**Comment by vdimir** [comment]
> Integration tests (asan, analyzer) [2/6] — fail: 1, passed: 318 [Details](https://s3.amazonaws.com/clickhouse-test-reports/56264/decf85baca2bdba6ad24f6f60775037deca54685/integration_tests__asan__analyzer__[2_6].html)

https://github.com/ClickHouse/ClickHouse/issues/55090


> Integration tests (tsan) [6/6] — fail: 1, passed: 490 [Details](https://s3.amazonaws.com/clickhouse-test-reports/56264/decf85baca2bdba6ad24f6f60775037deca54685/integration_tests__tsan__[6_6].html)

https://github.com/ClickHouse/ClickHouse/issues/56043

> Upgrade check (asan) — Logical error thrown (see clickhouse-server.log or logical_errors.txt) [Details](https://s3.amazonaws.com/clickhouse-test-reports/56264/decf85baca2bdba6ad24f6f60775037deca54685/upgrade_check__asan_.html)

https://github.com/ClickHouse/ClickHouse/issues/56202

**Pull request merged**